syntactic-paths/build.gradle

```diff
@@ -12,4 +13,33 @@ dependencies {
     testImplementation 'org.assertj:assertj-core'
     testImplementation 'org.junit.jupiter:junit-jupiter'
     testImplementation 'org.junit.jupiter:junit-jupiter-api'
+
+    jmhAnnotationProcessor 'org.openjdk.jmh:jmh-generator-annprocess'
+    jmh 'com.fasterxml.jackson.core:jackson-annotations'
+}
+
+jmh {
+    // Use profilers to collect additional data. Supported profilers:
+    // [cl, comp, gc, jfr, stack, perf, perfnorm, perfasm, xperf, xperfasm, hs_cl, hs_comp, hs_gc, hs_rt, hs_thr]
+    //profilers = ['jfr']
+    //jvmArgsPrepend = ['-XX:+UnlockDiagnosticVMOptions', '-XX:+DebugNonSafepoints']
+    humanOutputFile = project.file("${resultsDir(project)}/results.txt")
+    resultsFile = project.file("${resultsDir(project)}/results.json")
+    resultFormat = 'JSON'
+}
+
+tasks.jmhCompileGeneratedClasses {
+    options.annotationProcessorPath = configurations.errorprone
+    options.errorprone.enabled = true
 }
+
+
+def resultsDir(Project project) {
+    String circleArtifactsDir = System.getenv("CIRCLE_ARTIFACTS");
+    if (circleArtifactsDir == null) {
+        return project.getBuildDir().toString() + "/results/jmh";
+    }
+    return circleArtifactsDir + "/jmh/" + project.getName();
```

Comment:
we'll output the JMH results into [Circle's artifacts](https://app.circleci.com/pipelines/github/palantir/syntactic-paths/53/workflows/f8dc26d4-0326-4d6b-875f-21d0076d12ba/jobs/135/artifacts) for longer term comparison and can use https://jmh.morethan.io/ to easily compare the `results.json`

Reply:
This doesn't appear to work based on https://app.circleci.com/pipelines/github/palantir/syntactic-paths/59/workflows/2fb24d75-d8cb-4f77-bb51-ac4127a2c3ae/jobs/148/artifacts

Reply:
I suspect we may need a `.mkdirs()` somewhere

Reply:
The results from those are first two files under `~/artifacts/jmh/syntactic-paths/`:

- [~/artifacts/jmh/syntactic-paths/results.json](https://output.circle-artifacts.com/output/job/bb7409c0-4c53-43ac-b50b-fff9ff2ef85a/artifacts/0/~/artifacts/jmh/syntactic-paths/results.json)
- [~/artifacts/jmh/syntactic-paths/results.txt](https://output.circle-artifacts.com/output/job/bb7409c0-4c53-43ac-b50b-fff9ff2ef85a/artifacts/0/~/artifacts/jmh/syntactic-paths/results.txt)

Reply:
:facepalm: I see, sorry about that!

Reply:
no worries